catalyst_voices/packages/internal/catalyst_voices_services/lib/src/sync/sync_manager.dart

```diff
@@ -11,212 +11,241 @@ final _logger = Logger('SyncManager');
+
+  /// returns a list of [DocumentsSyncRequest] currently have recurring timer.
+  Iterable<DocumentsSyncRequest> get scheduledRequests;
+
+  /// A future that completes when the [activeRequest] finishes.
+  Future<bool> get waitForActiveRequest;
+
+  /// Removes [request] from pending and scheduled lists.
+  void cancel(DocumentsSyncRequest request);
 
-  /// Initializes the [SyncManager], expected to be called once per instance lifetime.
-  Future<void> init();
+  /// Adds [request] to the top of processing queue.
+  ///
+  /// Returned future will complete when [request] is processed.
+  Future<bool> complete(DocumentsSyncRequest request);
 
-  Future<void> start();
+  /// Cleans up resources, cancels timers, and closes streams.
+  Future<void> dispose();
+
+  /// Adds a [request] to the processing queue.
+  void queue(DocumentsSyncRequest request);
 }
 
 final class SyncManagerImpl implements SyncManager {
-  final AppMetaStorage _appMetaStorage;
+  final DocumentsSynchronizer _synchronizer;
   final SyncStatsStorage _statsStorage;
-  final DocumentsService _documentsService;
-  final CampaignService _campaignService;
   final CatalystProfiler _profiler;
 
   final _lock = Lock();
-  final _progressController = StreamController<double>.broadcast();
+  final _activeSyncProgressSC = StreamController<double>.broadcast();
+
+  final _requestsQueue = Queue<DocumentsSyncRequest>();
+  final _scheduledRequestsTimers = <DocumentsSyncRequest, Timer>{};
+  final _requestsCompleters = <DocumentsSyncRequest, Completer<bool>>{};
 
-  StreamSubscription<Campaign?>? _activeCampaignSub;
+  DocumentsSyncRequest? _activeRequest;
 
-  Timer? _syncTimer;
-  var _synchronizationCompleter = Completer<bool>();
+  bool _isDisposed = false;
 
   SyncManagerImpl(
-    this._appMetaStorage,
+    this._synchronizer,
     this._statsStorage,
-    this._documentsService,
-    this._campaignService,
     this._profiler,
   );
 
   @override
-  Stream<double> get progressStream => _progressController.stream;
+  DocumentsSyncRequest? get activeRequest => _activeRequest;
 
   @override
-  Future<bool> get waitForSync => _synchronizationCompleter.future;
+  Stream<double> get activeRequestProgress => _activeSyncProgressSC.stream;
 
   @override
-  Future<void> dispose() async {
-    await _activeCampaignSub?.cancel();
-    _activeCampaignSub = null;
-
-    _syncTimer?.cancel();
-    _syncTimer = null;
+  Iterable<DocumentsSyncRequest> get pendingRequests => List.unmodifiable(_requestsQueue);
 
-    if (!_synchronizationCompleter.isCompleted) {
-      _synchronizationCompleter.complete(false);
-    }
+  @override
+  Iterable<DocumentsSyncRequest> get scheduledRequests {
+    return List.unmodifiable(_scheduledRequestsTimers.keys);
+  }
 
-    await _progressController.close();
+  @override
+  Future<bool> get waitForActiveRequest {
+    final activeRequest = _activeRequest;
+    return _requestsCompleters[activeRequest]?.future ?? Future.value(false);
```

Comment:
Not sure if capturing it in local scpe is needed if we can use it directly: 
```suggestion
    return _requestsCompleters[_activeRequest]?.future ?? Future.value(false);
```